### PR TITLE
MINIFICPP-2026 Make isRunning member functions const

### PR DESCRIPTION
**Pull request by lordgamez**
https://issues.apache.org/jira/browse/MINIFICPP-2026

-----------------------
Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFICPP-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically main)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI results for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Review by szaszm — APPROVED** [review]

**Review by martinzink — APPROVED** [review]

File: extensions/libarchive/FocusArchiveEntry.h [hunk 1]
```diff
@@ -64,7 +64,7 @@ class FocusArchiveEntry : public core::Processor {
    public:
     explicit ReadCallback(core::Processor*, utils::file::FileManager *file_man, ArchiveMetadata *archiveMetadata);
     int64_t operator()(const std::shared_ptr<io::InputStream>& stream) const;
-    bool isRunning() {return proc_->isRunning();}
+    bool isRunning() const {return proc_->isRunning();}
```
Comment (fgerlits) [inline extensions/libarchive/FocusArchiveEntry.h:67]: should this be marked `override`, as well?

File: extensions/coap/server/CoapServer.h [hunk 2]
```diff
@@ -183,7 +183,7 @@ class CoapServer : public core::Connectable {
   /**
    * Determines if we are connected and operating
    */
-  virtual bool isRunning() {
+  virtual bool isRunning() const {
```
Comment (fgerlits) [inline extensions/coap/server/CoapServer.h:186]: I think this should be `override` instead of `virtual`

File: libminifi/include/c2/C2Protocol.h [hunk 3]
```diff
@@ -75,7 +75,7 @@ class C2Protocol : public core::Connectable {
   /**
    * Determines if we are connected and operating
    */
-  virtual bool isRunning() {
+  virtual bool isRunning() const {
```
Comment (fgerlits) [inline libminifi/include/c2/C2Protocol.h:78]: `virtual` -> `override`

File: libminifi/include/controllers/ThreadManagementService.h [hunk 4]
```diff
@@ -91,7 +91,7 @@ class ThreadManagementService : public core::controller::ControllerService {
   void yield() {
   }
 
-  bool isRunning() {
+  bool isRunning() const {
```
Comment (fgerlits) [inline libminifi/include/controllers/ThreadManagementService.h:94]: I think this should be marked `override`, as well

**Review by fgerlits — COMMENTED** [review]
not strictly related to constness, but since we're touching these lines, we might as well fix the `override` attributes, too

File: extensions/libarchive/FocusArchiveEntry.h (hunk 1 above)
Comment (lordgamez) [reply extensions/libarchive/FocusArchiveEntry.h]: This method was unused so I removed it in a20c3a130d80bd73e4383b8161e23c52c77fb1fd

File: extensions/coap/server/CoapServer.h (hunk 2 above)
Comment (lordgamez) [reply extensions/coap/server/CoapServer.h]: Updated in a20c3a130d80bd73e4383b8161e23c52c77fb1fd

File: libminifi/include/c2/C2Protocol.h (hunk 3 above)
Comment (lordgamez) [reply libminifi/include/c2/C2Protocol.h]: Updated in a20c3a130d80bd73e4383b8161e23c52c77fb1fd

File: libminifi/include/controllers/ThreadManagementService.h (hunk 4 above)
Comment (lordgamez) [reply libminifi/include/controllers/ThreadManagementService.h]: Updated in a20c3a130d80bd73e4383b8161e23c52c77fb1fd

**Review by fgerlits — APPROVED** [review]

**Pull request closed without merging**